homeassistant/components/soma/__init__.py

```diff
@@ -99,3 +106,38 @@ def device_info(self) -> DeviceInfo:
             manufacturer="Wazombi Labs",
             name=self.name,
         )
+
+    def set_position(self, position: int) -> None:
+        """Set the current device position."""
+        self.current_position = position
+        self.schedule_update_ha_state()
+
+    async def async_get_state_from_api(self) -> dict:
+        """Get the latest data from the API."""
+        response = {}
+        try:
+            response_from_api = await self.hass.async_add_executor_job(
+                self.api.get_shade_state, self.device["mac"]
+            )
+        except RequestException:
+            if self.api_is_available:
+                _LOGGER.warning("Connection to SOMA Connect failed")
+                self.api_is_available = False
+        else:
+            if not self.api_is_available:
+                self.api_is_available = True
+                _LOGGER.info("Connection to SOMA Connect succeeded")
+
+            if not is_api_response_success(response_from_api):
+                if self.is_available:
+                    self.is_available = False
+                    _LOGGER.warning(
+                        f'Device is unreachable ({self.name}). Error while fetching the state: {response_from_api["msg"]}'
```

Comment:
For loggers we do want to use % formatting.  Sorry if it is a bit confusing. 

logger = % formatting
everything else = f-strings

Reply:
Hehe no problem. Thanks for the info
Fixed